crates/wasi-http/src/streams_impl.rs

```diff
@@ -62,14 +75,30 @@ impl crate::streams::Host for WasiHttp {
         buf: Vec<u8>,
     ) -> wasmtime::Result<Result<u64, StreamError>> {
         match self.streams.get(&this) {
-            Some(data) => {
+            Some(st) => {
+                if st.closed {
+                    bail!("stream is dropped!");
+                }
+                let data = &st.data;
                 let mut new = bytes::BytesMut::with_capacity(data.len() + buf.len());
                 new.put(data.clone());
                 new.put(bytes::Bytes::from(buf.clone()));
-                self.streams.insert(this, new.freeze());
+                self.streams.insert(
+                    this,
+                    Stream {
+                        closed: false,
+                        data: new.freeze(),
+                    },
+                );
             }
             None => {
-                self.streams.insert(this, bytes::Bytes::from(buf.clone()));
+                self.streams.insert(
+                    this,
+                    Stream {
+                        closed: false,
+                        data: bytes::Bytes::from(buf.clone()),
```

Comment:
You wouldn't need this clone if the call to `buf.len` below was moved higher, before the match.

Reply:
done.